src/store/archiveJobs/actions.js

```diff
@@ -0,0 +1,41 @@
+import { currentAddressSelector } from "src/store"
+import { daoByType } from "src/store";
+import { JOB_STATE_FINISHED, JOB_STATE_FINALIZED } from 'src/models'
+import * as profileApi from 'src/api/backend'
+
+export const SELECT_INITIAL_PROPS_REQUEST = 'ARCHIVE_JOBS/SELECT_INITIAL_PROPS/REQUEST'
+export const SELECT_INITIAL_PROPS_SUCCESS = 'ARCHIVE_JOBS/SELECT_INITIAL_PROPS/SUCCESS'
+export const SELECT_INITIAL_PROPS_FAILURE = 'ARCHIVE_JOBS/SELECT_INITIAL_PROPS/FAILURE'
+export const selectInitialPropsRequest = (req) => ({ type: SELECT_INITIAL_PROPS_REQUEST, payload: req })
+export const selectInitialPropsSuccess = (res) => ({ type: SELECT_INITIAL_PROPS_SUCCESS, payload: res })
+export const selectInitialPropsFailure = (err) => ({ type: SELECT_INITIAL_PROPS_FAILURE, payload: err })
+export const selectInitialProps = () => async (dispatch, getState) => {
+  try {
+    dispatch(selectInitialPropsRequest())
+    const allowedStatuses = [JOB_STATE_FINISHED, JOB_STATE_FINALIZED]
+    const state = getState()
+    const JobsDataProvider = daoByType('JobsDataProvider')(state)
+    const BoardController = daoByType('BoardController')(state)
+    const userAddress = currentAddressSelector()(state)
+    const jobs = (await JobsDataProvider.getJobs(BoardController)).filter((x) => x.boardId)
+    const boards = await BoardController.getBoards(userAddress)
+    const userJobs = jobs.filter((x) => x.client.toLowerCase() === userAddress.toLowerCase())
+
+    const workerAddresses = userJobs
+      .reduce((result, x) => (x.worker ? result.concat(x) : result), [])
+      .map((x) => x.worker)
```

Comment:
the order of filtration must be like this
```
const clientJobs = jobs.filter((x) => x.client.toLowerCase() === userAddress.toLowerCase())
const clientArchiveJobs = clientJobs.filter((x) => allowedStatuses.includes(x.state))
const workerAddresses = clientArchiveJobs.map((x) => x.worker)
const workerPersons = await profileApi.getPersons(workerAddresses)
```

if you do filtration like this then you dont need to check if job has worker `.reduce((result, x) => (x.worker ? result.concat(x) : result), [])`